### ISSUE TITLE
Optimize CUDA data transfer

### ISSUE THREAD

**Issue by bigaidream** [nicholas-leonard/drmad#13]
[CUDA version](https://github.com/nicholas-leonard/drmad/blob/master/cuda_drmad_mnist.lua) is not optimized. It transfers data from CPU to GPU for every mini-batch. 

I'm not sure about the best practice when transferring data between CPU and GPU.

**Comment by nicholas-leonard** [comment]
@bigaidream I am working to modify `lr_drmad_mnist.lua` to have a `--cuda` cmd-line option to cast the experiment to cuda. That way, we only need to manage one script (or at least, one less script to manage).

**Comment by nicholas-leonard** [comment]
#14

**Comment by bigaidream** [comment]
It seems in `lr_drmad_mnist.lua`, the model does not learn anything. The accuracy is always 10%. 
I compared the code with previous one, but did not find the bug...

**Comment by nicholas-leonard** [comment]
Yeah I know. When I run it on my laptop, it works fine. On the server, it works 50% of the time. I am still looking into this.

**Comment by bigaidream** [comment]
No need at this moment. Focusing on the new API for learning rates, weight decay and momentum.
